processor/src/test/java/org/mapstruct/ap/test/targetthis/CustomerEntityMapper1.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.CollectionMappingStrategy;
+import org.mapstruct.InheritInverseConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(
+        unmappedSourcePolicy = ReportingPolicy.IGNORE,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED
+)
+public interface CustomerEntityMapper1 {
+    CustomerEntityMapper1 INSTANCE = Mappers.getMapper( CustomerEntityMapper1.class );
+
+    @Mapping(target = ".", source = "entity.*")
+    CustomerEntity map(CustomerDTO customer);
+
+    @InheritInverseConfiguration(name = "map")
+    // @Mapping(target = "entity", source = ".")
+    CustomerDTO inverseMap(CustomerEntity customer);
+
+    @Mapping(target = "entity", source = ".")
+    void update(CustomerEntity customer, @MappingTarget CustomerDTO dto);
+
+    @Mapping(target = ".", source = "entity.*")
+    void update(CustomerDTO dto, @MappingTarget CustomerEntity customer);
+
+    @Mapping(target = ".", source = "customer.entity.*")
```

Comment:
I don't think the `*` should be needed: `source = "customer.entity" ` should be sufficient. Although it might actually reflect better what you intent, it would also mean inconsistency with the current solution. Name base mapping is applied even without any `@Mapping` annotation. In the future we could even consider to reserve the `.*` for doing scanning on deeper nesting.

Reply:
I played around with this.. I you leave out the source altogether you get an NPE in annotation processing.. That cannot be the intention.

Reply:
> I don't think the * should be needed: source = "customer.entity"

But if I have <source = "customer.entity"> then mapper will try to set entity object to current target <target,setEntity(entity)>, right? While desire is to map every property from [source] entity to current target without need of naming every property.